### PR TITLE
Fix Issue 6585: Add offset to struct field arg

### PR DESCRIPTION
**Pull request by CarolEidt**
When the child of a `GT_OBJ` is a lclFldAddr, `genConsumePutStructArgStk()`
was not adding the offset of the struct field to the base address of
the local.

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/codegenxarch.cpp [hunk 1]
```diff
@@ -5280,56 +5280,56 @@ void CodeGen::genConsumePutStructArgStk(GenTreePutArgStk* putArgNode, regNumber
     // The putArgNode children are always contained. We should not consume any registers.
     assert(putArgNode->gtGetOp1()->isContained());
 
-    GenTree* dst = putArgNode;
+    GenTree* dstAddr = putArgNode;
```
Comment (BruceForstall) [inline src/jit/codegenxarch.cpp]: in the function header, this line needs editing: "baseVarNum - the base for var numfor placing the "by-value" args on the stack."

**Comment by BruceForstall** [comment]
Thanks for making it easier to read!

LGTM

**Comment by briansull** [comment]
LGTM as well

File: src/jit/codegenxarch.cpp (hunk 1 above)
Comment (CarolEidt) [inline src/jit/codegenxarch.cpp]: Done.

**Comment by mellinoe** [comment]
I'd love to see if this fixes the csc issue I was encountering on Ubuntu 16.10/openSUSE 42.1, but unfortunately an automatic windows update happened and I'm not able to start any of my virtualbox VM's anymore :disappointed: . @janvorli Are you able to test this against the csc repro I shared with you?

**Comment by janvorli** [comment]
@mellinoe, I am going to try it now.

**Comment by CarolEidt** [comment]
@janvorli - thanks!

**Comment by janvorli** [comment]
I have tried it - but I am hitting another unrelated issue now. So I cannot really tell now. I will try it in the morning.

**Comment by CarolEidt** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by janvorli** [comment]
I have locally fixed the other issue and I can confirm that this change fixed the original issue.

**Pull request merged**